packages/hebao_v1/contracts/base/MetaTxModule.sol

```diff
@@ -154,12 +154,12 @@ abstract contract MetaTxModule is BaseModule
     ///      Important! This function needs to be safe against re-entrancy by using
     ///      the 'Checks Effects Interactions' pattern! We do not use `nonReentrant`
     ///      because this function is used to call into the same contract.
+    ///
     /// @param data The raw transaction to be performed on this module.
     /// @param nonce The nonce of this meta transaction. When nonce is 0, this module will
     ///              make sure the transaction's metaTxHash is unique; otherwise, the module
     ///              requires the nonce is greater than the last nonce used by the same
     ///              wallet, but not by more than `block.number * 2^128`.
-    ///
     /// @param gasSetting A list that contains `gasToken` address, `gasPrice`, `gasLimit`,
```

Comment:
this can be passed in as a struct?

Reply:
Yeah, sure. If I remember correctly web3 still had some problem with encoding `BN`s in structs pretty recently, but I'm pretty sure that's fixed now (otherwise we'd also have this problem in protocol 3.5).

Reply:
Not a big deal, I think.